File: 1732._Find_the_Highest_Altitude/HighestAltitude.java
class HighestAltitude {
    int highestAltitude(int[] arr) {
        int maxAltitude = 0;
        int totalGain = 0;
        for (int x : arr) {
            totalGain += x;
            maxAltitude = Math.max(maxAltitude, totalGain);
        }
        return maxAltitude;
    }

    public static void main(String[] args) {
        HighestAltitude ha = new HighestAltitude();
        int[] arr = { -5, 1, 5, 0, -7 };
        System.out.println(ha.highestAltitude(arr));
    }
}

File: 1732._Find_the_Highest_Altitude/highestAltitude.js
const highestAltitude = (arr) => {
  let maxAltitude = 0;
  let totalGain = 0;
  for (const gain of arr) {
    totalGain += gain;
    maxAltitude = Math.max(maxAltitude, totalGain);
  }
  return maxAltitude;
};

console.log(highestAltitude([-5,1,5,0,-7]));